<!DOCTYPE html>

<html xmlns="http://www.w3.org/1999/xhtml">

<head>

<meta charset="utf-8" />
<meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
<meta name="generator" content="pandoc" />


<meta name="author" content="Hao He" />

<meta name="date" content="2018-10-04" />

<title>recombinations_in_DO_Liver_MEGAMUGA</title>

<script src="site_libs/jquery-1.12.4/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<script src="site_libs/htmlwidgets-1.3/htmlwidgets.js"></script>
<link href="site_libs/datatables-css-0.0.0/datatables-crosstalk.css" rel="stylesheet" />
<script src="site_libs/datatables-binding-0.4/datatables.js"></script>
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.min.css" rel="stylesheet" />
<link href="site_libs/dt-core-1.10.16/css/jquery.dataTables.extra.css" rel="stylesheet" />
<script src="site_libs/dt-core-1.10.16/js/jquery.dataTables.min.js"></script>
<link href="site_libs/crosstalk-1.0.0/css/crosstalk.css" rel="stylesheet" />
<script src="site_libs/crosstalk-1.0.0/js/crosstalk.min.js"></script>
<link href="site_libs/font-awesome-4.5.0/css/font-awesome.min.css" rel="stylesheet" />

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>


</head>

<body>

<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
  height: auto;
}
.tabbed-pane {
  padding-top: 12px;
}
button.code-folding-btn:focus {
  outline: none;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}

.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>


<div class="container-fluid main-container">

<!-- tabsets -->
<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});
</script>

<!-- code folding -->




<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}


.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
  padding-left: 25px;
  text-indent: 0;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>

<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">Workflowr_GBRS_array</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/xhyuo/Workflowr_GBRS_array">
    <span class="fa fa-github"></span>
     
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">recombinations_in_DO_Liver_MEGAMUGA</h1>
<h4 class="author"><em>Hao He</em></h4>
<h4 class="date"><em>2018-10-04</em></h4>

</div>


<p><strong>Last updated:</strong> 2018-10-04</p>
<strong>workflowr checks:</strong> <small>(Click a bullet for more information)</small>
<ul>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>R Markdown file:</strong> up-to-date </summary></p>
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Environment:</strong> empty </summary></p>
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Seed:</strong> <code>set.seed(20181001)</code> </summary></p>
<p>The command <code>set.seed(20181001)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Session information:</strong> recorded </summary></p>
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</details>
</li>
<li>
<p><details> <summary> <strong style="color:blue;">✔</strong> <strong>Repository version:</strong> <a href="https://github.com/xhyuo/Workflowr_GBRS_array/tree/04f52f61139197e8cb7513fef448884034c371f6" target="_blank">04f52f6</a> </summary></p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility. The version displayed above was the version of the Git repository at the time these results were generated. <br><br> Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
<pre><code>
Ignored files:
    Ignored:    .Rproj.user/

Untracked files:
    Untracked:  new_ini/
    Untracked:  output/DO_Liver/recomb.MM.genoprobs.HMM.RData
    Untracked:  output/DO_Liver/recomb.MM.genoprobs.RData
    Untracked:  output/DO_Liver/recomb.MM.genoprobs.array.RData
    Untracked:  output/DO_Liver/recomb.MUGA.genoprobs.RData

Unstaged changes:
    Modified:   analysis/recombinations_in_DO_Liver.Rmd

</code></pre>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes. </details>
</li>
</ul>
<details> <summary> <small><strong>Expand here to see past versions:</strong></small> </summary>
<ul>
<table style="border-collapse:separate; border-spacing:5px;">
<thead>
<tr>
<th style="text-align:left;">
File
</th>
<th style="text-align:left;">
Version
</th>
<th style="text-align:left;">
Author
</th>
<th style="text-align:left;">
Date
</th>
<th style="text-align:left;">
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
Rmd
</td>
<td style="text-align:left;">
<a href="https://github.com/xhyuo/Workflowr_GBRS_array/blob/04f52f61139197e8cb7513fef448884034c371f6/analysis/recombinations_in_DO_Liver_MEGAMUGA.Rmd" target="_blank">04f52f6</a>
</td>
<td style="text-align:left;">
xhyuo
</td>
<td style="text-align:left;">
2018-10-04
</td>
<td style="text-align:left;">
Publish
</td>
</tr>
</tbody>
</table>
</ul>
<p></details></p>
<hr />
<div id="load-packages" class="section level1">
<h1>Load packages</h1>
</div>
<div id="load-megamuga-raw-data-to-calculate-genoprobs" class="section level1">
<h1>Load megamuga raw data to calculate genoprobs</h1>
<pre class="r"><code>#megamuga
megamuga &lt;- list()
x0 &lt;- list()
y0 &lt;- list()
for(i in 1:3){
  megamuga[[i]] &lt;- read.table(paste0(&quot;/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/megamuga&quot;,i,&quot;.txt&quot;), header = TRUE, sep = &quot; &quot;)
  rownames(megamuga[[i]]) &lt;- megamuga[[i]]$marker
  #x
  x_col &lt;- grepl(&quot;_X&quot;,colnames(megamuga[[i]]))
  x0[[i]] &lt;- as.data.frame(megamuga[[i]][,x_col])
  colnames(x0[[i]]) &lt;- gsub(&#39;_X&#39;, &#39;&#39;, colnames(x0[[i]]))
  #y
  y_col &lt;- grepl(&quot;_Y&quot;,colnames(megamuga[[i]]))
  y0[[i]] &lt;- as.data.frame(megamuga[[i]][,y_col])
  colnames(y0[[i]]) &lt;- gsub(&#39;_Y&#39;, &#39;&#39;, colnames(y0[[i]]))
}
stopifnot(sum(rownames(x0[[1]]) != rownames(x0[[2]])) ==0)
stopifnot(sum(rownames(x0[[2]]) != rownames(x0[[3]])) ==0)
#combine x0 y0 list
x &lt;- as.matrix(t(do.call(cbind.data.frame,x0)))
y &lt;- as.matrix(t(do.call(cbind.data.frame,y0)))

# GBRS_sample
load(&quot;./output/DO_Liver/recomb.GBRS.genoprobs.RData&quot;)
GBRS_sample &lt;- as.character(recomb.GBRS.genoprobs$`0.110`$Sample)
# overlap samples between GBRS and megamuga (200 samples)
GBRS_MM_sample &lt;- intersect(GBRS_sample,rownames(x))

#subset GBRS_MM_sample
x &lt;- x[GBRS_MM_sample,]
y &lt;- y[GBRS_MM_sample,]

#megamuga
load(url(&quot;ftp://ftp.jax.org/MUGA/MM_snps.Rdata&quot;))
sex = sex.predict(x = x, y = y, snps = MM_snps, plot = T)
#sample sheet
sample_sheet &lt;- read.table(&quot;/projects/heh/HH/GBRS_array/data/DO_Liver/sample_sheet.tsv&quot;, header = TRUE)
#get match samples
match_sample_sheet &lt;- sample_sheet[sample_sheet$Sample %in% rownames(x),]
match_sample_sheet &lt;- match_sample_sheet[match(rownames(x),match_sample_sheet$Sample),]
#get generation from sample sheet
gen = paste(&quot;DO&quot;, match_sample_sheet$Gen, sep = &quot;&quot;)
names(gen) = as.character(match_sample_sheet$Sample)
#sex
sex &lt;- as.character(match_sample_sheet$Sex)
names(sex) = as.character(match_sample_sheet$Sample)

gen = gen[names(gen) %in% names(sex)]
gen = gen[match(names(sex), names(gen))]

stopifnot(all(rownames(x) == names(sex)))
stopifnot(all(rownames(x) == names(gen)))
# 200 samples and 77725 markers
data = list(x = x, y = y, sex = sex, gen = gen)
save(data, file = &quot;/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MEGAMUGA/MEGAMUGA.data.GBRSoverlapsample.RData&quot;)
#genotype probs
calc.genoprob(data = data, chr = &quot;all&quot;, output.dir = &quot;/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MEGAMUGA/&quot;, array = &quot;megamuga&quot;,sampletype = &quot;DO&quot;, method = &quot;intensity&quot;)

# GBRS_MM_sample genoprobs
mm.genoprobs &lt;- list()
for (i in GBRS_MM_sample){
  mm.genoprobs[[i]] &lt;- get(load(paste0(&quot;/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MEGAMUGA/&quot;,i,&quot;.genotype.probs.Rdata&quot;)))
}
mm.genoprobs.3d &lt;- abind(mm.genoprobs, along = 3)
#mm.genoprobs.3d 3D, subjects*genotypes*markers
mm.genoprobs.3d &lt;- aperm(mm.genoprobs.3d, perm = c(3,2,1)
#megamuga snps
load(url(&quot;ftp://ftp.jax.org/MUGA/MM_snps.Rdata&quot;))
subset.MM.snps &lt;- MM_snps[MM_snps$marker %in% dimnames(mm.genoprobs.3d)[[3]],]
chr &lt;- unique(subset.MM.snps$chr)
#split in chromosome
mm.genoprobs.3dchr &lt;- list()
for (c in chr){
  chr.marker &lt;- subset.MM.snps[subset.MM.snps$chr == c,&quot;marker&quot;]
  mm.genoprobs.3dchr[[c]] &lt;- mm.genoprobs.3d[,,chr.marker]
}
attr(mm.genoprobs.3dchr, &quot;crosstype&quot;) &lt;- &quot;do&quot;
save(mm.genoprobs.3dchr, file = &quot;/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MEGAMUGA/mm.genoprobs.array.3dchr.RData&quot;)

#use 100 cores
cl &lt;- makeCluster(100)
registerDoParallel(cl)
getDoParWorkers()
#For each individual at each position, find the genotype with the maximum marginal probability.
g &lt;- maxmarg(mm.genoprobs.3dchr,cores = 100)
#count_xo: Count numbers of crossovers in qtl2
n_xo &lt;- count_xo(g)
tot_n_xo &lt;- apply(n_xo,1,sum)
#make results as a dataframe
recomb.MM.genoprobs &lt;- data.frame(Sample = dimnames(mm.genoprobs.3dchr$`1`)[[1]],
     recomb = tot_n_xo)
#save
save(recomb.MM.genoprobs, file = &quot;./output/DO_Liver/recomb.MM.genoprobs.array.RData&quot;)</code></pre>
</div>
<div id="recombination-plot-between-gbrs-and-megamuga" class="section level1">
<h1>recombination plot between GBRS and MEGAMUGA</h1>
<pre class="r"><code>load(&quot;./output/DO_Liver/recomb.GBRS.genoprobs.RData&quot;)
load(&quot;./output/DO_Liver/recomb.MM.genoprobs.array.RData&quot;)
#remove outlier
remove = which(recomb.MM.genoprobs$recomb &gt; 1000)
#15 outliers
recomb.MM.genoprobs[remove,]
#     Sample recomb
# 3     F326   4685
# 39    F362   1962
# 40    F363   2315
# 45    F368   2607
# 101    F96   6391
# 167   M387   1469
# 168   M388   7848
# 172   M392  22657
# 173   M393   4539
# 174   M394  23450
# 178   M398   1405
# 184   M405  23455
# 187   M408  17512
# 190   M411   3147
# 195   M419  24556
#15 outliers
MM.outlier &lt;- as.character(recomb.MM.genoprobs[remove,]$Sample)
recomb.MM.genoprobs &lt;- recomb.MM.genoprobs[-remove,]
#sample sheet
sample_sheet &lt;- read.table(&quot;/projects/heh/HH/GBRS_array/data/DO_Liver/sample_sheet.tsv&quot;, header = TRUE)
#add gen
#merge with sample sheet
recomb.MM.plot &lt;- merge(recomb.MM.genoprobs, sample_sheet, by = &quot;Sample&quot;, all.x = TRUE)
recomb.MM.plot$Gen &lt;- as.factor(recomb.MM.plot$Gen)
#get summary table for each generation for 185 samples
summary.MM.recomb &lt;- ddply(recomb.MM.plot,.(Gen),
                           summarise,mean = mean(recomb),var = var(recomb), n= length(recomb))
#boxplot
p1 &lt;-ggplot(recomb.MM.plot, aes(x = Gen, y = recomb)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.25), aes(colour = Gen)) +
  theme(legend.position=&quot;none&quot;) +
  geom_smooth(method = &quot;lm&quot;, se=FALSE, color=&quot;black&quot;, aes(group=1), linetype=&quot;dotted&quot;) +
  labs(x = &quot;DO Outbreeding Generations&quot;) +
  labs(y = paste0(&quot;Number of Recombinations (MUGA)&quot;)) +
  scale_x_discrete(limits=c(&quot;4&quot;,&quot;5&quot;,&quot;6&quot;,&quot;7&quot;,&quot;8&quot;,&quot;9&quot;,&quot;10&quot;,&quot;11&quot;)) +
  ggtitle(&quot;Recombinations using genoprobs from MEGAMUGA samples (185)&quot;) +
  theme(plot.title = element_text(hjust = 0.5))
print(p1)</code></pre>
<p><img src="figure/recombinations_in_DO_Liver_MEGAMUGA.Rmd/recombination%20plot%20between%20GBRS%20and%20MEGAMUGA-1.png" width="3840" style="display: block; margin: auto;" /></p>
<pre class="r"><code>
#boxplot for GBRS
sigma = c(&quot;0.110&quot;, &quot;0.115&quot;, &quot;0.120&quot;, &quot;0.125&quot;, &quot;0.130&quot;)
p &lt;- list()
summary.GBRS.recomb &lt;- list()
for (sg in sigma){
  #merge with sample sheet
  recomb.GBRS.plot &lt;- merge(recomb.GBRS.genoprobs[[sg]], sample_sheet, by = &quot;Sample&quot;, all.x = TRUE)
  subset.recomb.GBRS.plot &lt;- recomb.GBRS.plot[recomb.GBRS.plot$Sample %in% recomb.MM.plot$Sample, ]
  subset.recomb.GBRS.plot$Gen &lt;- as.factor(subset.recomb.GBRS.plot$Gen)
  
  #get summary table for each generation per sigma value
  summary.GBRS.recomb[[sg]] &lt;- ddply(subset.recomb.GBRS.plot,.(Gen),
                           summarise,mean = mean(recomb),var = var(recomb), n= length(recomb))
  summary.GBRS.recomb[[sg]]$sigma &lt;- sg

  #boxplot
  p[[sg]]&lt;- ggplot(subset.recomb.GBRS.plot, aes(x = Gen, y = recomb)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(position = position_jitter(0.25), aes(colour = Gen)) +
    theme(legend.position=&quot;none&quot;) +
    geom_smooth(method = &quot;lm&quot;, se=FALSE, color=&quot;black&quot;, aes(group=1), linetype=&quot;dotted&quot;) +
    labs(x = &quot;DO Outbreeding Generations&quot;) +
    labs(y = paste0(&quot;Number of Recombinations (GBRS,sigma =&quot;, sg, &quot;)&quot;)) +
    scale_x_discrete(limits=c(&quot;4&quot;,&quot;5&quot;,&quot;6&quot;,&quot;7&quot;,&quot;8&quot;,&quot;9&quot;,&quot;10&quot;,&quot;11&quot;)) +
    ylim(150, 670)
}
#plot between GBRS and MEGAMUGA
grid.arrange(p1,p[[1]],p[[2]],p[[3]],p[[4]],p[[5]], ncol=6,top = textGrob(&quot;Recombination events between GBRS and MUGA&quot;,gp=gpar(fontsize=20,font=3)))</code></pre>
<p><img src="figure/recombinations_in_DO_Liver_MEGAMUGA.Rmd/recombination%20plot%20between%20GBRS%20and%20MEGAMUGA-2.png" width="3840" style="display: block; margin: auto;" /></p>
<pre class="r"><code>#get summary table for GBRS
summary.GBRS.recomb &lt;- do.call(rbind.data.frame,summary.GBRS.recomb)
rownames(summary.GBRS.recomb) &lt;- NULL
#display summary table summary.GBRS.recomb
DT::datatable(summary.GBRS.recomb,options = list(pageLength = 100, scrollY = &quot;800px&quot;, scrollX = &quot;20px&quot;))</code></pre>
<div id="htmlwidget-c6bffa569d5cccd793de" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-c6bffa569d5cccd793de">{"x":{"filter":"none","data":[["1","2","3","4","5","6","7","8","9","10","11","12","13","14","15"],["4","7","11","4","7","11","4","7","11","4","7","11","4","7","11"],[288.5,391.166666666667,494.960451977401,273.5,380.166666666667,476.446327683616,273.5,368.833333333333,462.553672316384,259.5,361.166666666667,450.768361581921,254.5,359.5,440.672316384181],[24.5,367.766666666667,683.720017976374,364.5,347.766666666667,629.237159732922,612.5,187.366666666667,563.418977914741,544.5,180.566666666667,523.928993323061,264.5,290.3,501.755649717514],[2,6,177,2,6,177,2,6,177,2,6,177,2,6,177],["0.110","0.110","0.110","0.115","0.115","0.115","0.120","0.120","0.120","0.125","0.125","0.125","0.130","0.130","0.130"]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gen<\/th>\n      <th>mean<\/th>\n      <th>var<\/th>\n      <th>n<\/th>\n      <th>sigma<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":100,"scrollY":"800px","scrollX":"20px","columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<pre class="r"><code>#display summary table summary.MM.recomb
DT::datatable(summary.MM.recomb,options = list(pageLength = 100, scrollY = &quot;100px&quot;, scrollX = &quot;20px&quot;))</code></pre>
<div id="htmlwidget-59869239fd7f888d78b8" style="width:100%;height:auto;" class="datatables html-widget"></div>
<script type="application/json" data-for="htmlwidget-59869239fd7f888d78b8">{"x":{"filter":"none","data":[["1","2","3"],["4","7","11"],[400.5,509,622.988700564972],[420.5,1060,3551.7271443246],[2,6,177]],"container":"<table class=\"display\">\n  <thead>\n    <tr>\n      <th> <\/th>\n      <th>Gen<\/th>\n      <th>mean<\/th>\n      <th>var<\/th>\n      <th>n<\/th>\n    <\/tr>\n  <\/thead>\n<\/table>","options":{"pageLength":100,"scrollY":"100px","scrollX":"20px","columnDefs":[{"className":"dt-right","targets":[2,3,4]},{"orderable":false,"targets":0}],"order":[],"autoWidth":false,"orderClasses":false}},"evals":[],"jsHooks":[]}</script>
<div id="session-information" class="section level2">
<h2>Session information</h2>
<pre class="r"><code>sessionInfo()
# R version 3.3.2 (2016-10-31)
# Platform: x86_64-pc-linux-gnu (64-bit)
# Running under: CentOS release 6.7 (Final)
# 
# locale:
#  [1] LC_CTYPE=en_US.UTF-8       LC_NUMERIC=C              
#  [3] LC_TIME=en_US.UTF-8        LC_COLLATE=en_US.UTF-8    
#  [5] LC_MONETARY=en_US.UTF-8    LC_MESSAGES=en_US.UTF-8   
#  [7] LC_PAPER=en_US.UTF-8       LC_NAME=C                 
#  [9] LC_ADDRESS=C               LC_TELEPHONE=C            
# [11] LC_MEASUREMENT=en_US.UTF-8 LC_IDENTIFICATION=C       
# 
# attached base packages:
#  [1] grid      stats4    parallel  stats     graphics  grDevices utils    
#  [8] datasets  methods   base     
# 
# other attached packages:
#  [1] DT_0.4                             plyr_1.8.4                        
#  [3] doParallel_1.0.10                  iterators_1.0.8                   
#  [5] foreach_1.4.3                      VennDiagram_1.6.17                
#  [7] futile.logger_1.4.3                gridExtra_2.2.1                   
#  [9] DOQTL_1.10.0                       VariantAnnotation_1.20.3          
# [11] Rsamtools_1.26.2                   SummarizedExperiment_1.4.0        
# [13] Biobase_2.34.0                     BSgenome.Mmusculus.UCSC.mm10_1.4.0
# [15] BSgenome_1.42.0                    rtracklayer_1.34.2                
# [17] Biostrings_2.42.1                  XVector_0.14.1                    
# [19] GenomicRanges_1.26.4               GenomeInfoDb_1.10.3               
# [21] IRanges_2.8.2                      S4Vectors_0.12.2                  
# [23] BiocGenerics_0.20.0                ggplot2_2.2.1                     
# [25] qtl2plot_0.5-14                    qtl2scan_0.5-22                   
# [27] qtl2geno_0.5-32                    qtl2_0.5-7                        
# [29] abind_1.4-5                        reticulate_1.10                   
# 
# loaded via a namespace (and not attached):
#  [1] bitops_1.0-6             bit64_0.9-7             
#  [3] rprojroot_1.3-2          prabclus_2.2-6          
#  [5] regress_1.3-15           tools_3.3.2             
#  [7] backports_1.0.5          R6_2.2.2                
#  [9] DBI_1.0.0                lazyeval_0.2.0          
# [11] colorspace_1.3-2         annotationTools_1.48.0  
# [13] trimcluster_0.1-2        nnet_7.3-12             
# [15] bit_1.1-14               git2r_0.23.0            
# [17] formatR_1.5              labeling_0.3            
# [19] diptest_0.75-7           scales_0.4.1            
# [21] QTLRel_1.0               DEoptimR_1.0-8          
# [23] mvtnorm_1.0-5            robustbase_0.92-7       
# [25] stringr_1.2.0            digest_0.6.13           
# [27] rmarkdown_1.9            R.utils_2.5.0           
# [29] htmltools_0.3.6          htmlwidgets_1.3         
# [31] rlang_0.2.2              RSQLite_2.1.1           
# [33] shiny_1.0.5              hwriter_1.3.2           
# [35] jsonlite_1.5             crosstalk_1.0.0         
# [37] gtools_3.5.0             mclust_5.2.1            
# [39] BiocParallel_1.8.2       R.oo_1.21.0             
# [41] RCurl_1.95-4.10          magrittr_1.5            
# [43] modeltools_0.2-21        qtl_1.41-6              
# [45] Matrix_1.2-14            Rcpp_0.12.18            
# [47] munsell_0.4.3            R.methodsS3_1.7.1       
# [49] stringi_1.1.2            whisker_0.3-2           
# [51] yaml_2.1.14              MASS_7.3-50             
# [53] zlibbioc_1.20.0          rhdf5_2.18.0            
# [55] flexmix_2.3-13           blob_1.1.1              
# [57] promises_1.0.1           gdata_2.18.0            
# [59] lattice_0.20-35          GenomicFeatures_1.26.2  
# [61] annotate_1.52.1          knitr_1.20              
# [63] pillar_1.0.1             RUnit_0.4.31            
# [65] fpc_2.1-10               corpcor_1.6.9           
# [67] codetools_0.2-15         biomaRt_2.30.0          
# [69] futile.options_1.0.1     XML_3.98-1.16           
# [71] qtl2convert_0.5-8        evaluate_0.10           
# [73] lambda.r_1.2.2           data.table_1.11.4       
# [75] httpuv_1.4.5             gtable_0.2.0            
# [77] kernlab_0.9-25           mime_0.5                
# [79] xtable_1.8-2             later_0.7.5             
# [81] qtl2db_0.5-5             class_7.3-14            
# [83] tibble_1.4.1             GenomicAlignments_1.10.1
# [85] AnnotationDbi_1.36.2     memoise_1.1.0           
# [87] workflowr_1.1.1          cluster_2.0.7-1</code></pre>
</div>
</div>

<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>

<hr>
<p>
  This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a>
  analysis was created with
  <a href="https://github.com/jdblischak/workflowr">workflowr</a> 1.1.1
</p>
<hr>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
