---
title: "recombinations_in_DO_Liver"
author: "Hao He"
date: "`r Sys.Date()`"
output: 
  rmarkdown::html_vignette:
  toc: true
vignette: >
  %\VignetteIndexEntry{recombinations_in_DO_Liver}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---
  
```{r knitr-opts, include=FALSE}
knitr::opts_chunk$set(comment = "#",collapse = TRUE)
```

```{r load_packages, include=FALSE}
library(reticulate)
library(abind)
library(qtl2)
library(ggplot2)
library(DOQTL)
library(gridExtra)
library(VennDiagram)
library(foreach)
library(doParallel)
library(parallel)
library(plyr)
library(DT)
```

# Load DO_liver data (gbrs.reconstructed.genotypes)
```{r load DO_liver data (gbrs.reconstructed.genotypes), message=FALSE, eval=FALSE}
# sigma and chr
chr.names <- c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","X","MT")
sigma = c("0.110", "0.115", "0.120", "0.125", "0.130")

# using .genotypes.npz file to count the number of recombination events
recomb.GBRS.genotypes <- list()
for (sg in sigma){
  sample.name <- list.files(path = paste0("/projects/heh/HH/GBRS_array/data/DO_Liver/results/bowtie-1.1.2_gbrs-0.1.6_R84-REL1505/sigma",sg,"/"))
  print(sg)
  recomb_each_sample <- list()
  for(i in sample.name){
    np <- reticulate::import("numpy")
    npz <- np$load(paste0("/projects/heh/HH/GBRS_array/data/DO_Liver/results/bowtie-1.1.2_gbrs-0.1.6_R84-REL1505/sigma",sg,"/", i, "/gbrs.reconstructed.genotypes.npz"))
    mx.state <- list()
    for(chr in chr.names){
      mx.state[[chr]] = as.numeric(as.factor(npz$f[[chr]]))
    }
    # Get the locations where the genotypes change.
    prox = lapply(mx.state, diff)
    prox = lapply(prox, "!=", 0)
    prox = lapply(prox, which)
    recomb_each_sample[[i]] <- sum(unlist(lapply(prox, length)))
  }
  recomb.GBRS.genotypes[[sg]] <- data.frame(Sample = names(recomb_each_sample), recomb = unlist(recomb_each_sample))
}
#save results
save(recomb.GBRS.genotypes, file = "./output/DO_Liver/recomb.GBRS.genotypes.RData")
```

# Load recombination results on recomb.GBRS.genotypes of DO_liver
```{r Load recombination results on recomb.GBRS.genotypes of DO_liver, message=FALSE, eval=TRUE, fig.height=10, fig.width=10}
load("./output/DO_Liver/recomb.GBRS.genotypes.RData")
#sample sheet
sample_sheet <- read.table("/projects/heh/HH/GBRS_array/data/DO_Liver/sample_sheet.tsv", header = TRUE)
sigma = c("0.110", "0.115", "0.120", "0.125", "0.130")
#get summary table for each generation per sigma value
summary.tab <- list()
for (sg in sigma){
  print(sg)
  #merge with sample sheet
  recomb.GBRS.genotypes.plot <- merge(recomb.GBRS.genotypes[[sg]], sample_sheet, by = "Sample", all.x = TRUE)
  recomb.GBRS.genotypes.plot$Gen <- as.factor(recomb.GBRS.genotypes.plot$Gen)
  #get summary table for each generation per sigma value
  summary.tab[[sg]] <- ddply(recomb.GBRS.genotypes.plot,.(Gen),
                summarise,mean = mean(recomb),var = var(recomb), n= length(recomb))
  summary.tab[[sg]]$sigma <- sg
  
  #boxplot
  p <-ggplot(recomb.GBRS.genotypes.plot, aes(x = Gen, y = recomb)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(position = position_jitter(0.25), aes(colour = Gen)) +
    theme(legend.position="none") +
    geom_smooth(method = "lm", se=FALSE, color="black", aes(group=1), linetype="dotted") +
    labs(x = "DO Outbreeding Generations (DO_Liver)") +
    labs(y = paste0("Number of Recombinations (sigma =", sg, ")")) +
    scale_x_discrete(limits=c("4","5","6","7","8","9","10","11")) +
    ggtitle("Recombinations using gbrs.reconstructed.genotypes") +
    theme(plot.title = element_text(hjust = 0.5)) +
    ylim(150, 550)
  print(p)
}
summary.tab.genotype <- do.call(rbind.data.frame,summary.tab)
rownames(summary.tab.genotype) <- NULL
#display summary table
DT::datatable(summary.tab.genotype,options = list(pageLength = 100, scrollY = "800px", scrollX = "20px"))
```

# Load DO_liver data gbrs.interpolated.genoprobs to get 3D genoprobs array
```{r load DO_liver data (gbrs.interpolated.genoprobs), message=FALSE, eval=FALSE}
# sigma
sigma = c("0.110", "0.115", "0.120", "0.125", "0.130")
# chr
chr.names <- c("1","2","3","4","5","6","7","8","9","10","11","12","13","14","15","16","17","18","19","X")

for (sg in sigma){
  sample.name <- list.files(path = paste0("/projects/heh/HH/GBRS_array/data/DO_Liver/results/bowtie-1.1.2_gbrs-0.1.6_R84-REL1505/sigma",sg,"/"))
  print(sg)
  d3.geno.prob <- list()
  for (chr in chr.names){
    print(chr)
    for(i in sample.name){
      np <- reticulate::import("numpy")
      npz <- np$load(paste0("/projects/heh/HH/GBRS_array/data/DO_Liver/results/bowtie-1.1.2_gbrs-0.1.6_R84-REL1505/sigma",sg,"/", i, "/gbrs.interpolated.genoprobs.npz"))
      d3.geno.prob[[chr]][[i]] <- npz$f[[chr]]
    }
    d3.geno.prob[[chr]] <- abind(d3.geno.prob[[chr]], along=3 )
    d3.geno.prob[[chr]] <- aperm(d3.geno.prob[[chr]], perm = c(3,1,2))
  }
  names(d3.geno.prob) <- chr.names
  attr(d3.geno.prob, "crosstype") <- "do"
  #d3.geno.prob three dimension subjects*genotypes*markers
  geno.probs.liver[[sg]] <- d3.geno.prob
}

# Save geno.probs.liver to a file
saveRDS(geno.probs.liver, file = "/projects/heh/HH/GBRS_array/data/DO_Liver./output/DO_Liver/geno.probs.liver.rds")
```

# Load 3D genoprobs array (geno.probs.liver) from DO_liver data to calculate recombination
```{r load DO_liver data (3D genoprobs array, geno.probs.liver), message=FALSE, eval=FALSE}
#load geno.probs.liver
geno.probs.liver <- readRDS("./output/DO_Liver/geno.probs.liver.rds")

#use 100 cores
cl <- makeCluster(100)
registerDoParallel(cl)
getDoParWorkers()

# using 3d array genotype probs to calculate the number of recombination events by qtl2 package
recomb.GBRS.genoprobs <- list()
for (sg in sigma){
  #For each individual at each position, find the genotype with the maximum marginal probability.
  g <- maxmarg(geno.probs.liver[[sg]],cores = 100)
  #count_xo: Count numbers of crossovers in qtl2
  n_xo <- count_xo(g)
  tot_n_xo <- apply(n_xo,1,sum)
  #make results as a dataframe
  recomb.GBRS.genoprobs[[sg]] <- data.frame(Sample = dimnames(geno.probs.liver[[sg]]$`1`)[[1]],
                                            recomb = tot_n_xo)
}
stopCluster(cl)
#save recomb.GBRS.genoprobs
save(recomb.GBRS.genoprobs, file = "./output/DO_Liver/recomb.GBRS.genoprobs.RData")
```

# Load recombination results on recomb.GBRS.genoprobs of DO_liver
```{r Load recombination results on recomb.GBRS.genoprobs of DO_liver, message=FALSE, eval=TRUE, fig.height=10, fig.width=10}
load("./output/DO_Liver/recomb.GBRS.genoprobs.RData")
#sample sheet
sample_sheet <- read.table("/projects/heh/HH/GBRS_array/data/DO_Liver/sample_sheet.tsv", header = TRUE)
sigma = c("0.110", "0.115", "0.120", "0.125", "0.130")
#get summary table for each generation per sigma value
summary.tab <- list()
for (sg in sigma){
  print(sg)
  #merge with sample sheet
  recomb.GBRS.genoprobs.plot <- merge(recomb.GBRS.genoprobs[[sg]], sample_sheet, by = "Sample", all.x = TRUE)
  recomb.GBRS.genoprobs.plot$Gen <- as.factor(recomb.GBRS.genoprobs.plot$Gen)
  #get summary table for each generation per sigma value
  summary.tab[[sg]] <- ddply(recomb.GBRS.genoprobs.plot,.(Gen),
                summarise,mean = mean(recomb),var = var(recomb), n= length(recomb))
  summary.tab[[sg]]$sigma <- sg
  
  #boxplot
  p <-ggplot(recomb.GBRS.genoprobs.plot, aes(x = Gen, y = recomb)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(position = position_jitter(0.25), aes(colour = Gen)) +
    theme(legend.position="none") +
    geom_smooth(method = "lm", se=FALSE, color="black", aes(group=1), linetype="dotted") +
    labs(x = "DO Outbreeding Generations (DO_Liver)") +
    labs(y = paste0("Number of Recombinations (sigma =", sg, ")")) +
    scale_x_discrete(limits=c("4","5","6","7","8","9","10","11")) +
    ggtitle("Recombinations using gbrs.interpolated.genoprobs") +
    theme(plot.title = element_text(hjust = 0.5)) +
    ylim(200, 600)
  print(p)
}
summary.tab.genoprobs <- do.call(rbind.data.frame,summary.tab)
rownames(summary.tab.genoprobs) <- NULL
#display summary table
DT::datatable(summary.tab.genoprobs,options = list(pageLength = 100, scrollY = "800px", scrollX = "20px"))
```

# Load muga raw data to calculate genoprobs
```{r load muga data, message=FALSE, eval=FALSE}
#muga
muga <- list()
x0 <- list()
y0 <- list()
for(i in 1:7){
  muga[[i]] <- read.table(paste0("/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/muga",i,".txt"), header = TRUE, sep = " ")
  rownames(muga[[i]]) <- muga[[i]]$marker
  #x
  x_col <- grepl("_X",colnames(muga[[i]]))
  x0[[i]] <- muga[[i]][,x_col,drop = F]
  colnames(x0[[i]]) <- gsub('_X', '', colnames(x0[[i]]))
  colnames(x0[[i]]) <- gsub('X9376.', '', colnames(x0[[i]]))
  
  #y
  y_col <- grepl("_Y",colnames(muga[[i]]))
  y0[[i]] <- muga[[i]][,y_col,drop = F]
  colnames(y0[[i]]) <- gsub('_Y', '', colnames(y0[[i]]))
  colnames(y0[[i]]) <- gsub('X9376.', '', colnames(y0[[i]]))
}
stopifnot(sum(rownames(x0[[1]]) != rownames(x0[[2]])) ==0)
stopifnot(sum(rownames(x0[[2]]) != rownames(x0[[3]])) ==0)
#combine x0 y0 list
x <- as.matrix(t(do.call(cbind.data.frame,x0)))
y <- as.matrix(t(do.call(cbind.data.frame,y0)))
#muga snps
load(url("ftp://ftp.jax.org/MUGA/muga_snps.Rdata"))
#muga
sex = sex.predict(x = x, y = y, snps = muga_snps, plot = T)
# There are  wild samples with outlier intensities.
# Remove them from all data.
remove = which(rowMeans(x, na.rm = T) + rowMeans(y, na.rm = T) > 2)
x = x[-remove,]
y = y[-remove,]
sex.p = sex.predict(x = x, y = y, snps = muga_snps, plot = T)
#sample sheet
sample_sheet <- read.table("/projects/heh/HH/GBRS_array/data/DO_Liver/sample_sheet.tsv", header = TRUE)
#get match samples
match_sample_sheet <- sample_sheet[sample_sheet$Sample %in% rownames(x),]
match_sample_sheet <- match_sample_sheet[match(rownames(x),match_sample_sheet$Sample),]
#get generation from sample sheet
gen = paste("DO", match_sample_sheet$Gen, sep = "")
names(gen) = as.character(match_sample_sheet$Sample)
#sex
sex <- as.character(match_sample_sheet$Sex)
names(sex) = as.character(match_sample_sheet$Sample)

gen = gen[names(gen) %in% names(sex)]
gen = gen[match(names(sex), names(gen))]

stopifnot(all(rownames(x) == names(sex)))
stopifnot(all(rownames(x) == names(gen)))
data = list(x = x, y = y, sex = sex, gen = gen)
save(data, file = "/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MUGA/MUGA.data.RData")
calc.genoprob(data = data, chr = "all", output.dir = "/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MUGA", array = "muga", sampletype = "DO", method = "intensity")
```

# Load megamuga raw data to calculate genoprobs
```{r load megamuga data, message=FALSE, eval=FALSE}
#megamuga
megamuga <- list()
x0 <- list()
y0 <- list()
for(i in 1:3){
  megamuga[[i]] <- read.table(paste0("/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/megamuga",i,".txt"), header = TRUE, sep = " ")
  rownames(megamuga[[i]]) <- megamuga[[i]]$marker
  #x
  x_col <- grepl("_X",colnames(megamuga[[i]]))
  x0[[i]] <- as.data.frame(megamuga[[i]][,x_col])
  colnames(x0[[i]]) <- gsub('_X', '', colnames(x0[[i]]))
  #y
  y_col <- grepl("_Y",colnames(megamuga[[i]]))
  y0[[i]] <- as.data.frame(megamuga[[i]][,y_col])
  colnames(y0[[i]]) <- gsub('_Y', '', colnames(y0[[i]]))
}
stopifnot(sum(rownames(x0[[1]]) != rownames(x0[[2]])) ==0)
stopifnot(sum(rownames(x0[[2]]) != rownames(x0[[3]])) ==0)
#combine x0 y0 list
x <- as.matrix(t(do.call(cbind.data.frame,x0)))
y <- as.matrix(t(do.call(cbind.data.frame,y0)))

# GBRS_sample
load("./output/DO_Liver/recomb.GBRS.genoprobs.RData")
GBRS_sample <- as.character(recomb.GBRS.genoprobs$`0.110`$Sample)
# overlap samples between GBRS and megamuga (200 samples)
GBRS_MM_sample <- intersect(GBRS_sample,rownames(x))

#subset GBRS_MM_sample
x <- x[GBRS_MM_sample,]
y <- y[GBRS_MM_sample,]

#megamuga
load(url("ftp://ftp.jax.org/MUGA/MM_snps.Rdata"))
sex = sex.predict(x = x, y = y, snps = MM_snps, plot = T)
#sample sheet
sample_sheet <- read.table("/projects/heh/HH/GBRS_array/data/DO_Liver/sample_sheet.tsv", header = TRUE)
#get match samples
match_sample_sheet <- sample_sheet[sample_sheet$Sample %in% rownames(x),]
match_sample_sheet <- match_sample_sheet[match(rownames(x),match_sample_sheet$Sample),]
#get generation from sample sheet
gen = paste("DO", match_sample_sheet$Gen, sep = "")
names(gen) = as.character(match_sample_sheet$Sample)
#sex
sex <- as.character(match_sample_sheet$Sex)
names(sex) = as.character(match_sample_sheet$Sample)

gen = gen[names(gen) %in% names(sex)]
gen = gen[match(names(sex), names(gen))]

stopifnot(all(rownames(x) == names(sex)))
stopifnot(all(rownames(x) == names(gen)))

data = list(x = x, y = y, sex = sex, gen = gen)
save(data, file = "/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MEGAMUGA/MEGAMUGA.data.GBRSoverlapsample.RData")
#genotype probs
calc.genoprob(data = data, chr = "all", output.dir = "/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MEGAMUGA/", array = "megamuga",sampletype = "DO", method = "intensity")
```

# MEGAMUGA samples genotype probability in dgatti folder
```{r MEGAMUGA samples genotype probability in dgatti folder, message=FALSE, eval=FALSE}
# MEGAMUGA samples genotype probability in dgatti folder
mm.i.dir = "/projects/heh/HH/dgatti/MegaMUGA/HMM/intensity/all/"

# Get sample IDs for each dataset. (288 samples)
mm.i.samples = dir(path = mm.i.dir, pattern = "^KLS.*Rdata$", full.names = T)
names(mm.i.samples) = gsub( "/projects/heh/HH/dgatti/MegaMUGA/HMM/intensity/all//KLS|\\.genotype\\.probs\\.Rdata",
                            "", mm.i.samples)

# overlap samples between GBRS and megamuga (187 samples)
GBRS_MM_sample <- intersect(GBRS_sample,names(mm.i.samples))

# GBRS_MM_sample genoprobs
mm.genoprobs.HMM <- list()
for (i in GBRS_MM_sample){
  mm.genoprobs.HMM[[i]] <- get(load(paste0("/projects/heh/HH/dgatti/MegaMUGA/HMM/intensity/all/KLS",i,".genotype.probs.Rdata")))
}
mm.genoprobs.HMM.3d <- abind(mm.genoprobs.HMM, along = 3)
#mm.genoprobs.3d 3D, subjects*genotypes*markers
mm.genoprobs.HMM.3d <- aperm(mm.genoprobs.HMM.3d, perm = c(3,2,1)
#megamuga snps
load(url("ftp://ftp.jax.org/MUGA/MM_snps.Rdata"))
subset.MM.snps <- MM_snps[MM_snps$marker %in% dimnames(mm.genoprobs.HMM.3d)[[3]],]
chr <- unique(subset.MM.snps$chr)
#split in chromosome
mm.genoprobs.3dchr <- list()
for (c in chr){
  chr.marker <- subset.MM.snps[subset.MM.snps$chr == c,"marker"]
  mm.genoprobs.3dchr[[c]] <- mm.genoprobs.HMM.3d[,,chr.marker]
}
attr(mm.genoprobs.3dchr, "crosstype") <- "do"
save(mm.genoprobs.3dchr, file = "/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MEGAMUGA/mm.genoprobs.3dchr.RData")

#use 100 cores
cl <- makeCluster(100)
registerDoParallel(cl)
getDoParWorkers()
#For each individual at each position, find the genotype with the maximum marginal probability.
g <- maxmarg(mm.genoprobs.3dchr,cores = 100)
#count_xo: Count numbers of crossovers in qtl2
n_xo <- count_xo(g)
tot_n_xo <- apply(n_xo,1,sum)
#make results as a dataframe
recomb.MM.genoprobs <- data.frame(Sample = dimnames(mm.genoprobs.3dchr$`1`)[[1]],
                                    recomb = tot_n_xo)
#save
save(GBRS_MM_sample, recomb.MM.genoprobs, file = "./output/DO_Liver/recomb.MM.genoprobs.RData")
```

# Load muga genotype probs into 3D and summarize the recombinations
```{r Load muga genotype probs into 3D and summarize the recombinations, message=FALSE, eval=FALSE}
#load muga raw data, data object
load("/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MUGA/MUGA.data.RData")
#muga sample name
muga.samplename <- rownames(data$x) #533 samples
#muga genoprobs
muga.genoprobs <- list()
for (i in muga.samplename){
  muga.genoprobs[[i]] <- get(load(paste0("/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MUGA/",i,".genotype.probs.Rdata")))
}
muga.genoprobs.3d <- abind(muga.genoprobs, along = 3)
#muga.genoprobs.3d 3D, subjects*genotypes*markers
muga.genoprobs.3d <- aperm(muga.genoprobs.3d, perm = c(3,2,1))
#muga snps
load(url("ftp://ftp.jax.org/MUGA/muga_snps.Rdata"))
subset.muga.snps <- muga_snps[muga_snps$marker %in% rownames(muga.genoprobs$F100),]
chr <- unique(subset.muga.snps$chr)
#split in chromosome
muga.genoprobs.3dchr <- list()
for (c in chr){
  chr.marker <- subset.muga.snps[subset.muga.snps$chr == c,"marker"]
  muga.genoprobs.3dchr[[c]] <- muga.genoprobs.3d[,,chr.marker]
}
attr(muga.genoprobs.3dchr, "crosstype") <- "do"
save(muga.genoprobs.3dchr, file = "/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MUGA/muga.genoprobs.3dchr.RData")
#use 100 cores
cl <- makeCluster(100)
registerDoParallel(cl)
getDoParWorkers()
#For each individual at each position, find the genotype with the maximum marginal probability.
g <- maxmarg(muga.genoprobs.3dchr,cores = 100)
#count_xo: Count numbers of crossovers in qtl2
n_xo <- count_xo(g)
tot_n_xo <- apply(n_xo,1,sum)
#make results as a dataframe
recomb.MUGA.genoprobs <- data.frame(Sample = dimnames(muga.genoprobs.3dchr$`1`)[[1]],
                                    recomb = tot_n_xo)
#summarize the recombinations by DOQTL
recomb.MUGA.doqtl = summarize.genotype.transitions(path = "/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MUGA", snps = muga_snps)
#save
save(recomb.MUGA.genoprobs, recomb.MUGA.doqtl, file = "./output/DO_Liver/recomb.MUGA.genoprobs.RData")
```

# Load recombinations on muga genotype probs data
```{r load muga genotype probs data, message=FALSE, eval=TRUE}
load("./output/DO_Liver/recomb.MUGA.genoprobs.RData")
#remove outlier
which(recomb.MUGA.genoprobs$Sample == "F138")
which(recomb.MUGA.genoprobs$Sample == "F96")
#47 109
recomb.MUGA.genoprobs <- recomb.MUGA.genoprobs[c(-47,-109),]
#sample sheet
sample_sheet <- read.table("/projects/heh/HH/GBRS_array/data/DO_Liver/sample_sheet.tsv", header = TRUE)
#add gen
#merge with sample sheet
recomb.muga.plot <- merge(recomb.MUGA.genoprobs, sample_sheet, by = "Sample", all.x = TRUE)
recomb.muga.plot$Gen <- as.factor(recomb.muga.plot$Gen)
#boxplot
p <-ggplot(recomb.muga.plot, aes(x = Gen, y = recomb)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.25), aes(colour = Gen)) +
  theme(legend.position="none") +
  geom_smooth(method = "lm", se=FALSE, color="black", aes(group=1), linetype="dotted") +
  labs(x = "DO Outbreeding Generations") +
  labs(y = paste0("Number of Recombinations (MUGA)")) +
  scale_x_discrete(limits=c("4","5","6","7","8")) +
  ggtitle("Recombinations using genoprobs from all MUGA samples") +
  theme(plot.title = element_text(hjust = 0.5)) +
  ylim(150, 700)
print(p)
```

# Venn diagram among GBRS, MUGA and MEGAMUGA
```{r Venn diagram among GBRS, MUGA and MEGAMUGA, message=FALSE, eval=TRUE,fig.height=30, fig.width=40}
#GBRS_sample
load("./output/DO_Liver/recomb.GBRS.genoprobs.RData")
GBRS_sample <- as.character(recomb.GBRS.genoprobs$`0.110`$Sample)
#muga_sample
load("/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MUGA/MUGA.data.RData")
muga_sample <- as.character(rownames(data$x))
#megamuga_sample
load("/projects/heh/HH/GBRS_array/data/DO_Liver/array_data/MEGAMUGA/MEGAMUGA.data.RData")
megamuga_sample <- as.character(rownames(data$x))

#count for venn diagram
area1 <- length(GBRS_sample)
area2 <- length(muga_sample)
area3 <- length(megamuga_sample)
n12 <- length(intersect(GBRS_sample,muga_sample))
n23 <- length(intersect(muga_sample,megamuga_sample))
n13 <- length(intersect(GBRS_sample,megamuga_sample))
n123 <-length(intersect(intersect(GBRS_sample,muga_sample),megamuga_sample))

#venn diagram between GBRS and MUGA
grid.newpage()
venn.plot <- draw.pairwise.venn(area1 = area1, area2 = area2, cross.area = n12, c("GBRS", "MUGA") , lty = "blank",
                 fill = c("skyblue", "pink1"), cex = 4, cat.cex = 5)

#venn diagram between GBRS and MEGAMUGA
grid.newpage()
venn.plot <- draw.pairwise.venn(area1 = area1, area2 = area3, cross.area = n13, c("GBRS", "MEGAMUGA"), lty = "blank", 
                 fill = c("skyblue", "mediumorchid"), cex = 4, cat.cex = 5)

#venn diagram between GBRS, MUGA and MEGAMUGA
grid.newpage()
draw.triple.venn(area1, area2, area3, n12, n23, n13, n123, 
                 category = c("GBRS", "MUGA", "MEGAMUGA"), lty = "blank", 
                 fill = c("skyblue", "pink1", "mediumorchid"), cex = 4, cat.cex = 5)
```

# recombination plot between GBRS and muga
```{r recombination plot between GBRS and muga, message=FALSE, eval=TRUE,fig.height=30, fig.width=40}
overlap_GBRS_muga <- intersect(GBRS_sample, muga_sample)
#overlap
subset.recomb.muga.plot <- recomb.muga.plot[recomb.muga.plot$Sample %in% overlap_GBRS_muga, ]
#get summary table for each generation
summary.muga.recomb <- ddply(subset.recomb.muga.plot,.(Gen),
                           summarise,mean = mean(recomb),var = var(recomb), n= length(recomb))

#boxplot for muga
p1 <-ggplot(subset.recomb.muga.plot, aes(x = Gen, y = recomb)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.25), aes(colour = Gen)) +
  theme(legend.position="none") +
  geom_smooth(method = "lm", se=FALSE, color="black", aes(group=1), linetype="dotted") +
  labs(x = "DO Outbreeding Generations") +
  labs(y = paste0("Number of Recombinations (MUGA)")) +
  scale_x_discrete(limits=c("4","5","6","7")) +
  ylim(150, 670)

#boxplot for GBRS
sigma = c("0.110", "0.115", "0.120", "0.125", "0.130")
p <- list()
summary.GBRS.recomb <- list()
for (sg in sigma){
  #merge with sample sheet
  recomb.GBRS.plot <- merge(recomb.GBRS.genoprobs[[sg]], sample_sheet, by = "Sample", all.x = TRUE)
  subset.recomb.GBRS.plot <- recomb.GBRS.plot[recomb.GBRS.plot$Sample %in% overlap_GBRS_muga, ]
  subset.recomb.GBRS.plot$Gen <- as.factor(subset.recomb.GBRS.plot$Gen)
  
  #get summary table for each generation per sigma value
  summary.GBRS.recomb[[sg]] <- ddply(subset.recomb.GBRS.plot,.(Gen),
                           summarise,mean = mean(recomb),var = var(recomb), n= length(recomb))
  summary.GBRS.recomb[[sg]]$sigma <- sg

  #boxplot
  p[[sg]]<- ggplot(subset.recomb.GBRS.plot, aes(x = Gen, y = recomb)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(position = position_jitter(0.25), aes(colour = Gen)) +
    theme(legend.position="none") +
    geom_smooth(method = "lm", se=FALSE, color="black", aes(group=1), linetype="dotted") +
    labs(x = "DO Outbreeding Generations") +
    labs(y = paste0("Number of Recombinations (GBRS,sigma =", sg, ")")) +
    scale_x_discrete(limits=c("4","5","6","7")) +
    ylim(150, 670)
}
#plot between GBRS and MUGA
grid.arrange(p1,p[[1]],p[[2]],p[[3]],p[[4]],p[[5]], ncol=6,top = textGrob("Recombination events between GBRS and MUGA",gp=gpar(fontsize=20,font=3)))
#get summary table for GBRS
summary.GBRS.recomb <- do.call(rbind.data.frame,summary.GBRS.recomb)
rownames(summary.GBRS.recomb) <- NULL
#display summary table summary.GBRS.recomb
DT::datatable(summary.GBRS.recomb,options = list(pageLength = 100, scrollY = "800px", scrollX = "20px"))
#display summary table summary.muga.recomb
DT::datatable(summary.muga.recomb,options = list(pageLength = 100, scrollY = "100px", scrollX = "20px"))
```

# recombination plot between GBRS and MEGAMUGA
```{r recombination plot between GBRS and MEGAMUGA, message=FALSE, eval=TRUE,fig.height=30, fig.width=40}
load("./output/DO_Liver/recomb.MM.genoprobs.RData")
#remove outlier
which(recomb.MM.genoprobs$Sample == "M387")
which(recomb.MM.genoprobs$Sample == "M398")
#162 169
recomb.MM.genoprobs <- recomb.MM.genoprobs[c(-162,-169),]
#sample sheet
sample_sheet <- read.table("/projects/heh/HH/GBRS_array/data/DO_Liver/sample_sheet.tsv", header = TRUE)
#add gen
#merge with sample sheet
recomb.MM.plot <- merge(recomb.MM.genoprobs, sample_sheet, by = "Sample", all.x = TRUE)
recomb.MM.plot$Gen <- as.factor(recomb.MM.plot$Gen)
#get summary table for each generation
summary.MM.recomb <- ddply(recomb.MM.plot,.(Gen),
                             summarise,mean = mean(recomb),var = var(recomb), n= length(recomb))
#boxplot
p1 <-ggplot(recomb.MM.plot, aes(x = Gen, y = recomb)) +
  geom_boxplot(outlier.shape = NA) +
  geom_jitter(position = position_jitter(0.25), aes(colour = Gen)) +
  theme(legend.position="none") +
  geom_smooth(method = "lm", se=FALSE, color="black", aes(group=1), linetype="dotted") +
  labs(x = "DO Outbreeding Generations") +
  labs(y = paste0("Number of Recombinations (MUGA)")) +
  scale_x_discrete(limits=c("4","5","6","7","8","9","10","11")) +
  ggtitle("Recombinations using genoprobs from 185 MEGAMUGA samples") +
  theme(plot.title = element_text(hjust = 0.5)) +
  ylim(150, 900)

#boxplot for GBRS
sigma = c("0.110", "0.115", "0.120", "0.125", "0.130")
p <- list()
summary.GBRS.recomb <- list()
for (sg in sigma){
  #merge with sample sheet
  recomb.GBRS.plot <- merge(recomb.GBRS.genoprobs[[sg]], sample_sheet, by = "Sample", all.x = TRUE)
  subset.recomb.GBRS.plot <- recomb.GBRS.plot[recomb.GBRS.plot$Sample %in% recomb.MM.plot$Sample, ]
  subset.recomb.GBRS.plot$Gen <- as.factor(subset.recomb.GBRS.plot$Gen)
  
  #get summary table for each generation per sigma value
  summary.GBRS.recomb[[sg]] <- ddply(subset.recomb.GBRS.plot,.(Gen),
                                     summarise,mean = mean(recomb),var = var(recomb), n= length(recomb))
  summary.GBRS.recomb[[sg]]$sigma <- sg
  
  #boxplot
  p[[sg]]<- ggplot(subset.recomb.GBRS.plot, aes(x = Gen, y = recomb)) +
    geom_boxplot(outlier.shape = NA) +
    geom_jitter(position = position_jitter(0.25), aes(colour = Gen)) +
    theme(legend.position="none") +
    geom_smooth(method = "lm", se=FALSE, color="black", aes(group=1), linetype="dotted") +
    labs(x = "DO Outbreeding Generations") +
    labs(y = paste0("Number of Recombinations (GBRS,sigma =", sg, ")")) +
    scale_x_discrete(limits=c("4","5","6","7","8","9","10","11")) +
    ylim(150, 670)
}
#plot between GBRS and MEGAMUGA
grid.arrange(p1,p[[1]],p[[2]],p[[3]],p[[4]],p[[5]], ncol=6,top = textGrob("Recombination events between GBRS and MEGAMUGA",gp=gpar(fontsize=20,font=3)))
#get summary table for GBRS
summary.GBRS.recomb <- do.call(rbind.data.frame,summary.GBRS.recomb)
rownames(summary.GBRS.recomb) <- NULL
#display summary table summary.GBRS.recomb
DT::datatable(summary.GBRS.recomb,options = list(pageLength = 100, scrollY = "800px", scrollX = "20px"))
#display summary table summary.MM.recomb
DT::datatable(summary.MM.recomb,options = list(pageLength = 100, scrollY = "100px", scrollX = "20px"))
```

